src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java

```diff
@@ -13,12 +34,478 @@
+        {
+            final AbstractBuilding buildingToDeliver = ((BuildingDeliveryman) ownBuilding).getBuildingToDeliver();
+            if (buildingToDeliver != null)
+            {
+                if (itemsToDeliver.isEmpty() && hasTools(buildingToDeliver))
+                {
+                    return DELIVERY;
+                }
+
+                gatherItems(buildingToDeliver);
+                setDelay(DUMP_AND_GATHER_DELAY);
+                return GATHER_IN_WAREHOUSE;
+            }
+        }
+        return START_WORKING;
+    }
+
+    /**
+     * Check if the deliveryman has all the tools to make the delivery.
+     *
+     * @param buildingToDeliver the building to deliver to.
+     * @return true if is ready to deliver.
+     */
+    private boolean hasTools(@NotNull final AbstractBuilding buildingToDeliver)
+    {
+        final String requiredTool = buildingToDeliver.getRequiredTool();
+        if (requiredTool.isEmpty())
+        {
+            return true;
+        }
+
+        if (requiredTool.equals(Utils.PICKAXE))
+        {
+            return InventoryUtils.isPickaxeInTileEntity(worker.getInventoryCitizen(), buildingToDeliver.getNeededPickaxeLevel());
+        }
+        return InventoryUtils.isToolInTileEntity(worker.getInventoryCitizen(), requiredTool, buildingToDeliver.getBuildingLevel());
+    }
+
+    /**
+     * Gather item from chest.
+     * Gathers only one stack of the item.
+     *
+     * @param buildingToDeliver building to deliver to.
+     */
+    private void gatherItems(@NotNull final AbstractBuilding buildingToDeliver)
+    {
+        BlockPos position;
+        if (itemsToDeliver.isEmpty())
+        {
+            final String tool = buildingToDeliver.getRequiredTool();
+            position = wareHouse.getTileEntity()
+                    .getPositionOfChestWithTool(tool,
+                            Utils.PICKAXE.equals(tool) ? buildingToDeliver.getNeededPickaxeLevel() : buildingToDeliver.getBuildingLevel(),
+                            buildingToDeliver);
+        }
+        else
+        {
+            final ItemStack stack = itemsToDeliver.get(0);
+            position = wareHouse.getTileEntity().getPositionOfChestWithItemStack(stack);
+        }
+
+        if (position == null)
+        {
+            return;
+        }
+
+        if (!worker.isWorkerAtSiteWithMove(position, MIN_DISTANCE_TO_CHEST))
+        {
+            return;
+        }
+
+        final TileEntity tileEntity = world.getTileEntity(position);
+        if (tileEntity instanceof TileEntityChest)
+        {
+            if(((TileEntityChest) tileEntity).numPlayersUsing == 0)
+            {
+                this.world.addBlockEvent(tileEntity.getPos(), tileEntity.getBlockType(), 1, 1);
+                this.world.notifyNeighborsOfStateChange(tileEntity.getPos(), tileEntity.getBlockType());
+                this.world.notifyNeighborsOfStateChange(tileEntity.getPos().down(), tileEntity.getBlockType());
+                setDelay(10);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Assign this magic number 10 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)